gateway/src/resty/limit/count-inc.lua

```diff
@@ -33,6 +33,17 @@ function _M.incoming(self, key, commit)
         if not count then
             return nil, err
         end
+
+        if count > limit then
+            count, err = dict:incr(key, -1)
```

Comment:
I believe this should be done by the `.uncommit` that is being called by `resty.limit.traffic`.

Reply:
@mikz you mean `.uncommit` needs to be called here with the condition of `i == n`, right?
https://github.com/openresty/lua-resty-limit-traffic/blob/master/lib/resty/limit/traffic.lua#L27-L29

However `resty.limit.conn` includes this.

So we should choose which file we edit, `resty.limit.traffic` and `resty.limit.conn`, or `resty.limit.count-inc`.

Reply:
@y-tabata I think we don't need to change any resty.limit library. We have to use it correctly.

`resty.limit.traffic` calls `:incoming(key, false)` until the last limiter which is called as `:incoming(key, true)`. If any of those return an error, it is fine to just exit. Nothing was committed to the database.

When all of them succeed, it tries to call again `:incoming(key, true)`, but in this case, if any of them fails then call `:uncommit(key)` for all limiters that it already called `:incoming`.

So I'd suspect our limiters are not rolled back because they are not wrapped in one `resty.limit.traffic`.

Reply:
> resty.limit.traffic calls :incoming(key, false) until the last limiter which is called as :incoming(key, true). If any of those return an error, it is fine to just exit. Nothing was committed to the database.

Using the current `count-inc`, I think that when the last limiter, which is called as :incoming(key, true), only returns nil and "rejected", the last one is committed to the database.

Reply:
Ok. I got it now. Implemented a fix to the upstream library: https://github.com/openresty/lua-resty-limit-traffic/pull/34/commits/c53f2240e953a9656580dbafcc142363ab063100

I think we can just copy that here.